web3/gas_strategies/time_based.py

```diff
@@ -55,6 +54,27 @@ def _get_avg_block_time(w3: Web3, sample_size: int) -> float:
     return (latest['timestamp'] - oldest['timestamp']) / constrained_sample_size
 
 
+def _get_weighted_avg_block_time(w3: Web3, sample_size: int) -> float:
```

Comment:
@pipermerriam do you have ideas on how we can test this? I did manually test it and confirm it works, but the eth-tester block times are all the same AFAICT.